actors/market/src/lib.rs

```diff
@@ -1049,6 +1050,37 @@ impl Actor {
         }
         Ok(())
     }
+
+    /// Fetches activation state for a deal.
+    /// This will be available from when the proposal is published until an undefined period after
+    /// the deal finishes (either normally or by termination).
+    /// Outside that period, returns USR_NOT_FOUND.
+    fn get_deal_activation(
+        rt: &mut impl Runtime,
+        params: GetDealActivationParams,
+    ) -> Result<GetDealActivationReturn, ActorError> {
+        rt.validate_immediate_caller_accept_any()?;
+        let found = rt.state::<State>()?.get_state(rt.store(), params.id)?;
+        match found {
+            // If we have state, the deal has been activated.
+            // It may also have completed normally, or been terminated, but not yet been cleaned up.
+            Some(state) => Ok(GetDealActivationReturn {
+                activated: state.sector_start_epoch,
+                terminated: state.slash_epoch,
+            }),
+            None => {
+                // With no state, the deal may have been published but not activated.
+                // If there's no proposal, we can't distinguish between never existing,
```

Comment:
So, we can be a little more clever than this. We can readily detect the "never existing" case -- it's simply `params.id >= st.next_id`. 

I agree that "failed to activate" vs. "cleaned up after completion/termination" are indistinguishable, though.